File: backend/api/app/Models/LeaveType.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class LeaveType extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'description',
        'default_quota',
        'is_active',
    ];

    protected $casts = [
        'default_quota' => 'integer',
        'is_active' => 'boolean',
    ];

    public function leaveBalances()
    {
        return $this->hasMany(LeaveBalance::class);
    }

    public function leaveRequests()
    {
        return $this->hasMany(LeaveRequest::class);
    }

    /**
     * Get active leave types
     */
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }
}

File: mobile/lib/models/leave_type.dart
class LeaveType {
  const LeaveType({
    required this.id,
    required this.name,
    this.description,
    this.defaultQuota,
  });

  final int id;
  final String name;
  final String? description;
  final int? defaultQuota;

  factory LeaveType.fromJson(Map<String, dynamic> json) {
    return LeaveType(
      id: json['id'] as int,
      name: json['name'] as String,
      description: json['description'] as String?,
      defaultQuota: json['default_quota'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'default_quota': defaultQuota,
    };
  }
}
